daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/Node.scala

```diff
@@ -179,6 +190,21 @@ object Node {
 
     override def requiredAuthorizers(): Set[Party] = key.maintainers
```

Comment:
It's a pity that we cannot use this method in authorization. Reason is mostly the backwards compatibility to transaction versions < 5, where we authorize a fetch node by checking that one of the stakeholders is authorizing. 
Should we consider dropping support for tx versions < 5? @andreaslochbihler-da @gerolf-da ?

Reply:
If you do drop support, you can also simplify `NodeFetch` to make actors non-optional.

Reply:
Yep, might be worth cleaning up support for the very old transaction versions. @gerolf-da is this feasible?

Reply:
See ["Permanent backward compatibility"](https://github.com/digital-asset/daml/blob/9d8e8d494a30b327db5483cdcbc0f489df468b32/daml-lf/governance.rst#permanent-backward-compatibility), and ["Queries" bullet point 6](https://github.com/digital-asset/daml/blob/9d8e8d494a30b327db5483cdcbc0f489df468b32/daml-lf/governance.rst#engine) for an explanation that "version _n_ was introduced" does not imply "version _n-1_ stopped being written".

Reply:
@S11001001 thanks for the links. Definitely had these things in mind and hence do not consider the PR in anyway ready yet. 

I'll split this PR into the informeesOfNode change and another one which actually brings in the correct semantics for it. If we want to go through with the "successful lookup by key becomes a fetch" and we want to retain full compatibility with old transaction versions we'll likely have no other choice then to introduce both a new transaction version and a new LF version (to use right semantics in LF to speedy compilation).